fluentd/lib/in_tail_metric/write_watcher.rb

```diff
@@ -0,0 +1,76 @@
+#
+# Fluentd
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+require 'fluent/plugin/input'
+
+module Fluent::Plugin
+  class TailInput < Fluent::Plugin::Input
```

Comment:
Need to rename class TailInputMetric to avoid clash with original plugin class.

Reply:
need to further discuss with you tonight.

Reply:
parent class class TailInput < Fluent::Plugin::Input not required so keeping only module Fluent::Plugin
  class WriteWatcher < Fluent::Plugin::Input